editor/animation_track_editor.cpp

```diff
@@ -4463,17 +4463,12 @@ void AnimationTrackEditor::insert_node_value_key(Node *p_node, const String &p_p
 	String path = root->get_path_to(p_node, true);
 
 	// Get the value from the subpath.
-	Variant value = p_node;
-	Vector<String> property_path = p_property.split(":");
-	for (const String &E : property_path) {
-		if (value.get_type() == Variant::OBJECT) {
-			Object *obj = value;
-			value = obj->get(E);
-		} else {
-			value = Variant();
-			break;
-		}
+	Vector<String> subpath = p_property.split(":");
+	Vector<StringName> subpath_names;
+	for (const String &s : subpath) {
+		subpath_names.push_back(s);
```

Comment:
I think the `p_property` string can be converted to `NodePath` and then these subpaths can be extracted with

```
Vector<StringName> subpath_names = p_property_path.get_as_property_path().get_subnames();
```